libs/agno/agno/models/response.py

```diff
@@ -113,6 +113,24 @@ class ModelResponse:
     extra: Optional[Dict[str, Any]] = None
 
 
+@dataclass
+class StreamData:
+    """ Cumulative result from a model stream """
+    response_role: Optional[Literal["system", "user", "assistant", "tool"]] = None
+    response_content: Any = ""
+    response_thinking: Any = ""
+    response_redacted_thinking: Any = ""
+    response_citations: Optional[Citations] = None
+    response_tool_calls: List[Dict[str, Any]] = field(default_factory=list)
+
+    response_audio: Optional[AudioResponse] = None
+    response_image: Optional[ImageArtifact] = None
+
+    # Data from the provider that we might need on subsequent messages
+    response_provider_data: Optional[Dict[str, Any]] = None
+
+    extra: Optional[Dict[str, Any]] = None
```

Comment:
What's response_provider_data? should we just have extra_data here?

Reply:
Yeah can do